src/docker.rs

```diff
@@ -1172,7 +1172,13 @@ impl Docker {
                     if !contents.is_empty() {
                         message = serde_json::from_str::<DockerServerErrorMessage>(&contents)
                             .map(|msg| msg.message)
-                            .or_else(|e| if e.is_data() { Ok(contents) } else { Err(e) })?;
+                            .or_else(|e| {
+                                if e.is_data() || e.is_syntax() {
+                                    Ok(contents)
+                                } else {
+                                    Err(e)
+                                }
+                            })?;
```

Comment:
@dsander It might be easier to test this as a unit test with the "yup-hyper-mock" crate to send a faulty json payload -- examples of this https://github.com/fussybeaver/bollard/blob/eb0ef0d121ab24ed6cf3adcc05fa35ba30625a6e/src/image.rs#L1516-L1545 , or test this function directly without all the other plumbing...

Reply:
Sorry, somehow I didn't see your previous message. Doing this as a unit test sounds like the better approach, Ill have a look at it in the next days, thanks for the hit!